sdk/src/org.graalvm.polyglot/src/org/graalvm/polyglot/Context.java

```diff
@@ -1180,6 +1183,30 @@ public Builder allowCreateThread(boolean enabled) {
             return this;
         }
 
+        /**
+         * Installs handler to control what happens on multiple thread access. When multiple threads
+         * are accessing a context which isn't ready for multithreaded access an exception is
+         * yielded by default. By installing this {@code handler} one can control what shall happen.
+         * Either to throw the provided exception or resolve the multithreaded situation and return
+         * to retry the thread access again.
+         * <p>
+         * To throw {@link SecurityException} instead of the default one use following snippet:
+         * {@snippet file="org/graalvm/polyglot/Context.java" region="throwSecurityException"}
+         * <p>
+         * To implement <em>"retry later on collision"</em> strategy use for example following
+         * snippet: {@snippet file="org/graalvm/polyglot/Context.java" region="retry"}
+         *
+         *
+         * @param handler callback that either throws the provided {@code RuntimeException} or
+         *            returns to signal request for retry
+         * @return this builder
+         * @since 24.2
+         */
+        public Builder onDeniedThreadAccess(Consumer<RuntimeException> handler) {
+            this.onDeniedThreadAccess = handler;
```

Comment:
I would feel more comfortable if this is only TruffleContext API for now.

Reply:
Done in e6e076e